### PR TITLE
labels: Add tracked/out-of-tree label to k/enhancements 

### PR DESCRIPTION
**Pull request by justaugustus**
I believe we discussed this idea in a recent SIG Release meeting.
This enables the Release Team's Enhancements subteam to tag an issue with `tracked/out-of-tree` to denote an issue should never be tracked by the team.

This PR also backfills a set of manually-created labels from the before times when I didn't know about `label_sync`:
- `stage/alpha`
- `stage/beta`
- `stage/stable`
- `tracked/no`
- `tracked/yes`

/sig release
/area enhancements release-team
/assign @mrbobbytables @jeremyrickard @johnbelamaric 
cc: @kubernetes/release-team 
Partial work for https://github.com/kubernetes/sig-release/issues/539 and https://github.com/kubernetes/sig-release/issues/486.

### CODE REVIEW

**Comment by mrbobbytables** [comment]
/lgtm

Thanks @justaugustus :)

File: label_sync/labels.yaml [hunk 1]
```diff
@@ -797,30 +797,67 @@ repos:
...
         description: Categorizes KEP tracking issues and PRs modifying the KEP directory
         name: kind/kep
         target: both
         prowPlugin: label
         addedBy: anyone
+      - color: d93f0b
+        description: Denotes an issue tracking an enhancement targeted for Alpha status
+        name: stage/alpha
+        target: issues
+        prowPlugin: label
```
Comment (palnabarun) [inline label_sync/labels.yaml:832]: I think the prow plugin for `stage/*` labels in `stage`

Ref: https://git.k8s.io/test-infra/prow/plugins/stage

File: label_sync/labels.yaml [hunk 2]
```diff
@@ -797,30 +797,67 @@ repos:
...
+        addedBy: anyone
+      - color: fbca04
+        description: Denotes an issue tracking an enhancement targeted for Beta status
+        name: stage/beta
+        target: issues
+        prowPlugin: label
```
Comment (palnabarun) [inline label_sync/labels.yaml:838]: Same comment as above.

File: label_sync/labels.yaml [hunk 3]
```diff
@@ -797,30 +797,67 @@ repos:
...
+        addedBy: anyone
+      - color: 0e8a16
+        description: Denotes an issue tracking an enhancement targeted for Stable/GA status
+        name: stage/stable
+        target: issues
+        prowPlugin: label
```
Comment (palnabarun) [inline label_sync/labels.yaml:844]: Same comment as above.

File: label_sync/labels.yaml (hunk 1 above)
Comment (mrbobbytables) [reply label_sync/labels.yaml:832]: You're right
https://github.com/kubernetes/test-infra/blob/d2e2e04d1b0335335ae006973ba606e95f585af9/prow/plugins/stage/stage.go#L32-L37

Comment (justaugustus) [reply label_sync/labels.yaml]: @palnabarun -- Nice catch. Fixed!

File: label_sync/labels.yaml (hunk 2 above)
Comment (justaugustus) [reply label_sync/labels.yaml]: https://github.com/kubernetes/test-infra/pull/17473#discussion_r418332532

File: label_sync/labels.yaml (hunk 3 above)
Comment (justaugustus) [reply label_sync/labels.yaml]: https://github.com/kubernetes/test-infra/pull/17473#discussion_r418332532

**Comment by justaugustus** [comment]
Addressed comments. Needs re-lgtm.

**Comment by palnabarun** [comment]
/lgtm

Thank you @justaugustus o/ :)

**Pull request merged**